internal/code/package.go

```diff
@@ -1,14 +1,68 @@
 package code
 
 import (
+	"errors"
 	"go/ast"
+	"go/parser"
+	"go/token"
 	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+var (
+	errNoPackagesFound = errors.New("no packages found")
 )
 
-// ParsePackage extracts package name, struct and interface implementations from
-// map[string]*ast.Package. Test files will be ignored.
-func ParsePackage(pkgs map[string]*ast.Package) (Package, error) {
+type parseDirFn func(dir string) (map[string]*ast.Package, error)
+
+type parsePackagePathFn func(dir string) (string, error)
+
+func defaultParseDir(dir string) (map[string]*ast.Package, error) {
+	return parser.ParseDir(token.NewFileSet(), dir, nil, parser.ParseComments)
+}
+func defaultPackagePathParser(dir string) (string, error) {
```

Comment:
just FYI

`mockgen` also have a func of same purpose. And it also refers to `packages.Load`. 

https://github.com/uber-go/mock/blob/7638c0a8373a607828ed93e62c0969140d5b0bc0/mockgen/mockgen.go#L860